### PR TITLE
Update the code with more POO

### PR DESCRIPTION
**Pull request by Benoit382**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -20,9 +20,19 @@
       "role": "Wrapper Developer"
     }
   ],
+  "require": {
+    "php": ">=7.2",
```
Comment (tazorax) [inline composer.json]: Can you change to PHP >= 7.1 ? It seems there are no usage of 7.2 features

File: src/Api.php [hunk 2]
```diff
@@ -0,0 +1,443 @@
+<?php
+
+namespace OpenFoodFacts;
+
+use OpenFoodFacts\Document;
```
Comment (tazorax) [inline src/Api.php]: these 2 imports are unused, please remove it

File: src/Collection.php [hunk 3]
```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace OpenFoodFacts;
+
+use OpenFoodFacts\Document;
```
Comment (tazorax) [inline src/Collection.php]: this import is unused, please remove it

File: src/Document.php [hunk 4]
```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace OpenFoodFacts;
+
+use OpenFoodFacts\Document\FoodProduct;
```
Comment (tazorax) [inline src/Document.php]: these 3 imports are unused, please remove it

File: src/Document.php [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    /**
+     * the whole data
+     * @var array
+     */
+    private $api;
+
+    /**
+     * Initialization the document and specify from which API it was extract
+     * @param array $data the whole data
+     * @param array $api the api name
```
Comment (tazorax) [inline src/Document.php]: typo mistake here 😉 : is string, isn't it ?

File: src/Api.php (hunk 2 above)
Comment (Benoit382) [reply src/Api.php]: The 4 imports in this class are used, 
Document: line 203
Collection : line 183
LoggerInterface: line 112
NullLogger: line 115

File: src/Collection.php (hunk 3 above)
Comment (Benoit382) [reply src/Collection.php]: This import is used at line 32.

File: src/Document.php (hunk 4 above)
Comment (Benoit382) [reply src/Document.php]: I will remove this imports.

File: src/Document.php (hunk 5 above)
Comment (Benoit382) [reply src/Document.php]: Yes.

File: src/Collection.php (hunk 3 above)
Comment (tazorax) [reply src/Collection.php]: My bad 😛

Comment (tazorax) [reply src/Collection.php]: Wait a minute, it's the same namespace, so no import is needed

File: src/Api.php (hunk 2 above)
Comment (tazorax) [reply src/Api.php]: Document and Collection is in the same namespace, no import required

Comment (Benoit382) [reply src/Api.php]: It's true, i didn't see it like that.

**Comment by tazorax** [comment]
@Benoit382 thanks a lot for your contribution !

**Pull request merged**